src/main/scala/org/llm4s/llmconnect/extractors/UniversalExtractor.scala

```diff
@@ -0,0 +1,18 @@
+package org.llm4s.llmconnect.extractors
+
+object UniversalExtractor {
+  def extract(pathOrUrl: String): String =
+    if (pathOrUrl.startsWith("http://") || pathOrUrl.startsWith("https://")) {
+      WebExtractor.extractText(pathOrUrl)
+    } else if (pathOrUrl.endsWith(".pdf")) {
+      PDFExtractor.extractText(pathOrUrl)
+    } else if (pathOrUrl.endsWith(".docx")) {
+      DocxExtractor.extractText(pathOrUrl)
+    } else if (pathOrUrl.endsWith(".xlsx")) {
+      ExcelExtractor.extractText(pathOrUrl)
+    } else if (pathOrUrl.endsWith(".txt")) {
```

Comment:
you could support a very comment type here - markdown (.md) just by making adding a rule for the text extractor.